### PR TITLE
Adding select component

### PR DESCRIPTION
**Pull request by samsmarthx**
#### What does this PR do? (please provide any background)
Extends select component to allow us to use the data attribute 

#### What tests does this PR have?
New 'select' test

#### How can this be tested?
`npm test`
`npm run docs`

#### Screenshots / Screencast
http://grab.by/M3S8

#### What gif best describes how you feel about this work?

---

- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md) and I'm happy for this to be reviewed.

#### Reviewers

**Review 1**
- [x] :+1:

**Review 2** \*
- [x] :+1:

**Review 3** _(optional)_
- [ ] :+1:

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

\*  for HX this review must be completed by an SE, SA or Project Guru

### CODE REVIEW

File: src/components/select/select.jsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    var classes = 'component-select';
+    var dataAttributes = this.getDataAttributesFromProps();
+
+    // the form label
+    var label;
+    if (this.props.label) {
+      label = ( <label className="component-select-label" htmlFor={this.props.id}>{this.props.label}</label > );
+    }
+
+    /**
```
Comment (dancork) [inline src/components/select/select.jsx]: I think error was for input validation, probably not needed for select

File: src/components/select/select.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  },
+
+  getDefaultProps: function() {
+    return {
+      id: 'component-select',
+      name: 'component-select'
+    };
+  },
+
+
```
Comment (dancork) [inline src/components/select/select.jsx]: Extra new line

File: docs/examples/Select.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+var example = (
+  <div>
+    <UIToolkit.Select label = 'Destination:'>
```
Comment (hxpaul) [inline docs/examples/Select.jsx]: would normally be `<UIToolkit.Select label="Destination:">`

File: src/components/select/select.jsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  handleChange: function(e) {
+    var self = this;
+    var value = (e.target) ? e.target.value : null;
```
Comment (hxpaul) [inline src/components/select/select.jsx]: unneccessary () there

File: src/components/select/select.jsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    if (self.props.handleChange) {
+      self.props.handleChange.apply(this, arguments);
+    }
+  },
+
+  render: function() {
+    var classes = 'component-select';
+    var dataAttributes = this.getDataAttributesFromProps();
+
+    // the form label
```
Comment (hxpaul) [inline src/components/select/select.jsx]: that comment doesn't add much imo

File: docs/examples/Select.jsx (hunk 3 above)
Comment (jodiedoubleday) [inline docs/examples/Select.jsx]: also remove the `:`

File: test/components/select-test.jsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      <SelectView label="test" />
+    );
+
+    var renderedSelect = TestUtils.findRenderedDOMComponentWithClass(select, 'component-select-label');
+    assert.isDefined(renderedSelect);
+
+  });
+
+  it('should not have default value', function() {
+    var select = TestUtils.renderIntoDocument(
```
Comment (Arditc) [inline test/components/select-test.jsx]: Can this be placed in an beforeEach? :)

File: src/components/select/select.jsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  propTypes: {
+    label: React.PropTypes.string,
+    name: React.PropTypes.string,
+    id: React.PropTypes.string,
+    children: React.PropTypes.string,
+    handleChange: React.PropTypes.func,
+    data: React.PropTypes.object
+  },
+
+  getInitialState: function() {
```
Comment (hxpaul) [inline src/components/select/select.jsx]: I can't figure out if this needs state or not

File: test/components/select-test.jsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    );
+    it('sets id to "component-select"', function() {
+      assert.equal(elem.props.id, 'component-select');
+    });
+
+    it('sets name to "component-select"', function() {
+      assert.equal(elem.props.name, 'component-select');
+    });
+  });
+
```
Comment (hxpaul) [inline test/components/select-test.jsx]: needs a test for handleChange

**Comment by hxpaul** [comment]
Taking review 2 - I think this will still work as you expect / need it to without any state at all. Also it needs a test for the change handler, apart from that this is really good.

File: test/components/select-test.jsx (hunk 8 above)
Comment (dancork) [inline test/components/select-test.jsx]: If state is removed so we need handleChange?

File: test/components/select-test.jsx (hunk 6 above)
Comment (samsmarthx) [inline test/components/select-test.jsx]: I don't think it can because it's rendered in differently in each case

**Comment by samsmarthx** [comment]
As @dancork suggested,  my plan is to add multiple select to this as well (https://react-bootstrap.github.io/components.html#input-types) I just wanted to get an idea of how I'd gotten on with the basics :)

**Comment by dancork** [comment]
Great work Sam, it's looking good

File: test/components/select-test.jsx (hunk 6 above)
Comment (Arditc) [inline test/components/select-test.jsx]: So you're not able to have e.g.

describe('when SelectView has no attributes', function() {
beforeEach(function() {
 var select = TestUtils.renderIntoDocument(
   <SelectView />
  );
 var renderedSelect = TestUtils.findRenderedDOMComponentWithClass(select, 'component-select-field');
});
 it('should not have default value', function() {
  assert.equal(renderedSelect.value, '');
 });
 it('should have default ID', function() {
 assert.equal(renderedSelect.id, 'component-select');
});
it('should have default name', function() {
assert.equal(renderedSelect.name, 'component-select');
 });
});

Comment (dancork) [inline test/components/select-test.jsx]: @Arditc in your example the scope of select and renderedSelect would be in the beforeEach anonymous function only.

Comment (samsmarthx) [inline test/components/select-test.jsx]: I could but testing that it has no attributes is not something I'm currently testing

Comment (dancork) [inline test/components/select-test.jsx]: If anything...

```
context('with no attributes', function() {
  var select = null;
  var renderedSelect = null;
  beforeEach(function() {
    select = TestUtils.renderIntoDocument(  <SelectView/ > );
    renderedSelect = TestUtils.findRenderedDOMComponentWithClass(select, 'component-select-field');
  });
  it('should not have default value', function() {
    assert.equal(renderedSelect.value, '');
  });
  it('should have default ID', function() {
    assert.equal(renderedSelect.id, 'component-select');
  });
  it('should have default name', function() {
    assert.equal(renderedSelect.name, 'component-select');
  });
});
```

File: src/components/select/select.jsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+      id: 'component-select',
+      name: 'component-select'
+    };
+  },
+
+  handleChange: function(e) {
+    var self = this;
+    var value = e.target ? e.target.value : null;
+
+    self.setState({
```
Comment (hxpaul) [inline src/components/select/select.jsx]: this can go to, don't need to set the state at all

File: test/components/select-test.jsx (hunk 8 above)
Comment (hxpaul) [inline test/components/select-test.jsx]: yes if you want to trigger something when it changes, pass it on back up the chain - the select doesn't need to maintain its own state though

**Comment by hxpaul** [comment]
Looks good to me. :+1: 
You would have been "at the helm" if you'd been the one to start using `expect` in the tests https://github.com/holidayextras/culture/blob/master/testing-standard.md#assertion-syntax but not a blocker for me, I think someone should go through and do the whole lot (probably me)

**Comment by dancork** [comment]
Lovely stuff :shipit:

**Comment by hxpaul** [comment]
oh one more thing you should update the version in package.json I think

**Pull request merged**